### DISCUSSION TITLE
feat: Streamlined Modelfile Creation with Local Model Selection and Extraction

### DISCUSSION THREAD

**Discussion by silentoplayz** [open-webui/open-webui#2239] [Ideas]
**Is your feature request related to a problem? Please describe.**
Users, particularly those without an existing Modelfile in the WebUI, may find it time-consuming to manually enter model details when creating their first Modelfile from scratch. While a clone feature is available, it requires users to have an existing modelfile in the `Modelfiles` tab before then gaining the ability to duplicate said Modelfile. This process can be time-consuming, as users currently lack the ability to directly create a new modelfile based on an existing, locally downloaded model without manual intervention.

**Describe the solution you'd like**
I propose an enhancement to the existing modelfile duplication feature, enabling users to extract relevant model details from an existing, locally downloaded model and auto-populate the modelfile creator tab, saving both time and effort. This enhancement will allow users to quickly create new modelfiles based on existing, locally downloaded models, making it easier to manage and maintain their modelfiles.

**Describe alternatives you've considered**
Two alternatives have been considered:

1. Viewing a model's details page from a model's tags page in ollama.com/library:
   - Go to a particular model's tags (e.g. <https://ollama.com/library/llama2/tags>).
   - Click on a tag (e.g. <https://ollama.com/library/llama2:13b>).
   - View the layers.
   - Note: If the FROM instruction is not present, it means the model was created from a local file.

2. Using the `ollama show` command to print the Modelfile for any local models:
   - Run `ollama show --modelfile <model-tag>`.
   - The Modelfile generated by "ollama show" can be used as a base for creating a new Modelfile.

However, these alternatives still require manual intervention and may not be as user-friendly as the proposed enhancement.

**Additional context**
The enhanced modelfile duplication feature with local model selection and modefile extraction will save users time by automating the process of filling in model details when creating a new modelfile from scratch. This will improve the user experience and streamline the process of managing and maintaining modelfiles.

**Comment by tjbck** [comment]
![image](https://github.com/open-webui/open-webui/assets/25473318/6eda164c-a582-4419-af8b-2bf58de2d293)

We already have a clone feature if that's what you meant.

**Comment by silentoplayz** [comment]
> ![image](https://private-user-images.githubusercontent.com/25473318/313360484-6eda164c-a582-4419-af8b-2bf58de2d293.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA1NDk3MDcsIm5iZiI6MTcxMDU0OTQwNywicGF0aCI6Ii8yNTQ3MzMxOC8zMTMzNjA0ODQtNmVkYTE2NGMtYTU4Mi00NDE5LWFmOGItMmJmNThkZTJkMjkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE2VDAwMzY0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwYzEyZDE2ODhlNzFhNTc1NTI0ZTM2MTViOWI4ZDgwYTRiZjg0NTMwMTFmYzUyOTgwZmJiY2FmZDcyNjdjMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.j7Rmpbd9a1fH2ECV0n8jjtQiPBtFfc8_9gn-4LFy5zE)
> 
> We already have a clone feature if that's what you meant.

While a clone feature is available and exists already after a modelfile has been created, it requires users to have an existing modelfile in the Modelfiles tab before having the ability to duplicate the modelfile itself. The enhanced model file duplication feature with local model selection and modelfile extraction proposes to save users time by automating the process of filling in model details when creating a new modelfile from scratch.

**Comment by silentoplayz** [comment]
## Update
Linking here for visibility - https://github.com/open-webui/open-webui/commit/b88f829dbb76842604b0c898be8023b0f40daeab
